src/core/experimental/memutils.d

```diff
@@ -0,0 +1,332 @@
+/**
+ * Pure D replacement of the C Standard Library basic memory building blocks of string.h
+ * Source: $(DRUNTIMESRC core/experimental/memutils.d)
+ */
+module core.experimental.memutils;
+
+unittest
+{
+    Dmemset_testStaticType!(byte)(5);
+    Dmemset_testStaticType!(ubyte)(5);
+    Dmemset_testStaticType!(short)(5);
+    Dmemset_testStaticType!(ushort)(5);
+    Dmemset_testStaticType!(int)(5);
+    Dmemset_testStaticType!(uint)(5);
+    Dmemset_testStaticType!(long)(5);
+    Dmemset_testStaticType!(ulong)(5);
+    Dmemset_testStaticType!(float)(5);
+    Dmemset_testStaticType!(double)(5);
+    Dmemset_testStaticType!(real)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 3);
+    static foreach (i; 1..10) {
+        Dmemset_testDynamicArray!(ubyte)(5, 2^^i);
+        Dmemset_testStaticArray!(ubyte, 2^^i)(5);
+    }
+    Dmemset_testDynamicArray!(ubyte)(5, 100);
+    Dmemset_testStaticArray!(ubyte, 100)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 500);
+    Dmemset_testStaticArray!(ubyte, 500)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 700);
+    Dmemset_testStaticArray!(ubyte, 700)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 3434);
+    Dmemset_testStaticArray!(ubyte, 3434)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 7128);
+    Dmemset_testStaticArray!(ubyte, 7128)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 13908);
+    Dmemset_testStaticArray!(ubyte, 13908)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 16343);
+    Dmemset_testStaticArray!(ubyte, 16343)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 27897);
+    Dmemset_testStaticArray!(ubyte, 27897)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 32344);
+    Dmemset_testStaticArray!(ubyte, 32344)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 46830);
+    Dmemset_testStaticArray!(ubyte, 46830)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 64349);
+    Dmemset_testStaticArray!(ubyte, 64349)(5);
+}
+
+// From a very good Chandler Carruth video on benchmarking: https://www.youtube.com/watch?v=nXaxk27zwlk
+void escape(void* p)
```

Comment:
We only need this for benchmarking, so this doesn't need to be part of this module. We have the https://github.com/dlang/druntime/tree/master/benchmark folder here (though these test run on linux only), but if you have any benchmark scripts they should be added to this folder. They don't have to be integrated with the suite, but they shouldn't be part of the released source code which they would if added here.

Reply:
I have not forgotten this one, I just picked other things first. The thing with escaping is that the reason we want it for benchmarking purposes is so that the compiler doesn't optimize it away.
But optimizing it away is a problem when testing as well. I don't remember the last time I checked _without_ escaping, when I have time I'll do and I'll post an update.

Reply:
Ok, escaping was removed. It's not needed and my best _guess_ is that this is because of the verification (if the code is optimized away, an assert will fire, which does not satisfy correctness).
Do you want to add benchmarks in `/benchmark` ?
I think that it's easier for people to do benchmarks using the main repo of the project. So maybe somehow redirect them there.